### ISSUE TITLE
review and refactor classes and modules

### ISSUE THREAD

**Issue by tonykgill** [cibolabs/pixeldriller#31]
Also consider renaming the project to pixeldriller. On reflection, I don't want to give the impression that it's tied to stac.

- main package is pixdrill
- Split pointstats.py into two modules: drillpoints.py and drillstats.py
- rename pixelstac.py to drill.py
- asset_reader.py remains unchanged

```
from pixdrill import drill
from pixdrill import drillpoints
from pixdrill import drillstats
from pixdrill import asset_reader
```

The pointsstats.py module is getting long. Split it into two modules, grouping similar functionality:
- drillpoints.py: classes Point, ItemPoints, ImageItem
- drillstats.py: the standard stats functions and the ItemStats class, although we may rename the class as discussed next

Modify how the stats for each point are stored. Rename ItemStats to PointStats. Move the handling of multiple sets of stats per point (one for each intersecting item) to this PointStats object (currently handled in the Point class). PointStats has two attributes: the point, and a dictionary of stats that is keyed by item ID. Each value in the dictionary is another dictionary keyed by the statistic name. Also add the Item to this dictionary using the "ITEM" key.

There is also an issue with how drill (currently pixelstac) calls calc_stats. If a point intersects multiple Items, then the stats for that item are repeatedly calculated on each call to calc_stats. A more efficient approach is to "calculate stats per point":
- loop over each item (using the ItemPoints lists), calling read_data(); this ensures images are opened once only
- then loop over each point calling calc_stats(), ensuring stats are calculated only once per item.

Document another procedures that users can follow in their own code, called "calculate stats per item":
- For each item (in the ItemPoints list):
  - call read_data()
  - then calc_stats()
  - then do something with the stats
  - then call reset()

The latter approach is demonstrated in cibo_etl.
